src/hooks/auth.ts

```diff
@@ -65,16 +68,30 @@ export const useAuth = ({ middleware }: { middleware?: any } = {}) => {
   const logout = async () => {
     await api.post('/api/logout');
     mutate(null);
+    destroyCookie(null, 'auth_token'); // Remove the auth token cookie
     router.push('/login');
   };
 
   useEffect(() => {
+    const cookies = parseCookies();
+
     if (user || error) {
       setIsLoading(false);
     }
 
     if (middleware == 'guest' && user) router.push('/campaign');
     if (middleware == 'auth' && error) router.push('/login');
+
+    // Check if auth token cookie exists
+    if (!user && !error && cookies.auth_token) {
+      setAuthToken(cookies.auth_token);
```

Comment:
shall we use camelCase as I see we're using it mostly on frontend?
```suggestion
      setAuthToken(cookies.authToken);
```